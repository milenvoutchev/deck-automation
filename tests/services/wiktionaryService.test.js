/* eslint-disable no-unused-vars */
import sinon from 'sinon';
import WiktionaryService from '../../server/services/wiktionaryService';
import mocks from '../mocks/wiktionary.json';

describe('wiktionaryService', () => {

  describe('isValidWordType()', () => {
    it('correctly recognises valid/invalid word types', () => {
      expect(WiktionaryService.isValidWordType('Verb')).toBeTruthy();
      expect(WiktionaryService.isValidWordType('Substantiv')).toBeTruthy();
      expect(WiktionaryService.isValidWordType('Adjektiv')).toBeTruthy();
      expect(WiktionaryService.isValidWordType('Deklinierte Form')).toBeFalsy();
    });
  });

  describe('createWordResearch()', () => {
    const wiktionaryService = new WiktionaryService();

    it('renders correctly against saved snapshots', () => {
      const { wikitext, categories } = mocks.rawData;
      const wordResearch = wiktionaryService.createWordResearch(wikitext, categories);

      // expect(JSON.stringify(wordResearch)).toBe(JSON.stringify(wiktionarySnapshots.expected));
      expect(wordResearch).toMatchSnapshot();
    });
  });
});
