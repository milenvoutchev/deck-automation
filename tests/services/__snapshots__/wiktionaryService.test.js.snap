// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wiktionaryService createWordResearch() renders correctly against saved snapshots 1`] = `
Object {
  "examples": undefined,
  "lautschrift": "ˈanɡəˌhɔɪ̯ɐt",
  "nounArticle": null,
  "nounGender": null,
  "nounPlural": null,
  "senses": Array [
    Object {
      "code": "[1]",
      "value": "''[[Nautik]]'': jemanden für den Dienst an Bord einstellen",
    },
    Object {
      "code": "[2]",
      "value": "''[[Nautik]]'': den Dienst an Bord antreten",
    },
    Object {
      "code": "[3]",
      "value": "jemanden für eine bestimmte Aufgabe gewinnen",
    },
  ],
  "sources": Object {
    "wikitext": "== anheuern ({{Sprache|Deutsch}}) ==
=== {{Wortart|Verb|Deutsch}} ===

{{Deutsch Verb Übersicht
|Präsens_ich=heuere an
|Präsens_du=heuerst an
|Präsens_er, sie, es=heuert an
|Präteritum_ich=heuerte an
|Partizip II=angeheuert
|Konjunktiv II_ich=heuerte an
|Imperativ Singular=heuere an
|Imperativ Plural=heuert an
|Hilfsverb=haben
}}

{{Worttrennung}}
:an·heu·ern, {{Prät.}} heu·er·te an, {{Part.}} an·ge·heu·ert

{{Aussprache}}
:{{IPA}} {{Lautschrift|ˈanˌhɔɪ̯ɐn}}, {{Prät.}} {{Lautschrift|ˌhɔɪ̯ɐtə ˈan}}, {{Part.}} {{Lautschrift|ˈanɡəˌhɔɪ̯ɐt}}
:{{Hörbeispiele}} {{Audio|De-anheuern.ogg}}, {{Prät.}} {{Audio|}}, {{Part.}} {{Audio|}}

{{Bedeutungen}}
:[1] ''[[Nautik]]'': jemanden für den Dienst an Bord einstellen
:[2] ''[[Nautik]]'': den Dienst an Bord antreten
:[3] jemanden für eine bestimmte Aufgabe gewinnen

{{Herkunft}}
:[1] [[Ableitung]] zu ''[[heuern]]'' mit dem [[Derivatem]] ([[Ableitungsmorphem]]) ''[[an-]]''

{{Gegenwörter}}
:[1, 2] [[abheuern]], [[abmustern]]
:[3] [[anwerben]]

{{Oberbegriffe}}
:[1] [[einstellen]]

{{Beispiele}}
:[1] „Sie ließen die Diligent komplett überholen und ''heuerten'' eine Mannschaft ''an'', welche die Waren für die erste Fahrt nach Martinique an Bord verstaute; das Ganze nahm beinahe neun Monate in Anspruch.“<ref>{{Literatur | Autor=Robert Harms| Titel =Das Sklavenschiff| TitelErg=Eine Reise in die Welt des Sklavenhandels| Verlag=C. Bertelsmann Verlag | Ort=ohne Ort | Jahr=2004 | ISBN=3-570-00277-2 |Seiten=124. }} Kursiv gedruckt: ''Diligent''.</ref>
:[2] Der Schiffsführer hat um 6 Uhr ''angeheuert.''
:[3] Für den Einbruch hat der Kriminelle zwei Komplizen ''angeheuert.''

==== {{Übersetzungen}} ====
{{Ü-Tabelle|Ü-links=
*{{en}}: [1] {{Ü|en|}}
*{{fr}}: [1] {{Ü|fr|}}
|Ü-rechts=
*{{ru}}: [1] {{Üt|ru||}}
*{{es}}: [1] {{Ü|es|}}
}}

{{Referenzen}}
:[1, 2] {{Ref-DWDS|anheuern}}
:[1, (3)] {{Ref-Canoo|anheuern}}
:[1] {{Ref-UniLeipzig|anheuern}}
:[1–3] {{Ref-FreeDictionary|anheuern}}
:[1–3] {{Ref-Duden|anheuern}}

{{Quellen}}",
  },
  "usages": Array [
    Object {
      "de": "„Sie ließen die Diligent komplett überholen und ''heuerten'' eine Mannschaft ''an'', welche die Waren für die erste Fahrt nach Martinique an Bord verstaute; das Ganze nahm beinahe neun Monate in Anspruch.“",
    },
    Object {
      "de": "Der Schiffsführer hat um 6 Uhr ''angeheuert.''",
    },
    Object {
      "de": "Für den Einbruch hat der Kriminelle zwei Komplizen ''angeheuert.''",
    },
  ],
  "verbPastParticiple": "angeheuert",
  "verbPerfectAuxiliaryThird": "hat",
  "verbPresentThirdPerson": "heuert an",
  "verbPreteriteFirstPerson": "heuerte an",
  "verbPreteriteThirdPerson": "heuerte an",
  "wordDe": "anheuern",
  "wordEn": undefined,
  "wordType": "Verb",
}
`;
